assignments/lambda-classes.js

```diff
@@ -1 +1,129 @@
 // CODE here for your Lambda Classes
+// Creating Person class
+class Person {
+  constructor(attr) {
+    this.name = attr.name;
+    this.age = attr.age;
+    this.location = attr.location;
+    this.gender = attr.gender;
+  }
+  speak() {
+    console.log(`Hello, my name is ${this.name}, I am from ${this.location}`);
+  }
+}
+
+class Instructor extends Person {
+  constructor(attr) {
+    super(attr);
+    this.specialty = attr.specialty;
+    this.favLanguage = attr.favLanguage;
+    this.catchPhrase = attr.catchPhrase;
+  }
+  demo(subject) {
+    console.log(`Today we are learning about ${subject}`);
+  }
+  grade(student, subject) {
+    console.log(`${student.name} receives a perfect score on ${subject}`);
+  }
+  hijinx(student) {
+    let luckyNum = Math.random() * (100);
+    let randomNum = Math.random() * (10);
+    if (luckyNum > 49) {
+      student.grade += randomNum;
+      console.log(`${student.name} gained ${this.name}'s favor and earned ${randomNum} points on their grade!`);
+    } else {
+      student.grade -= randomNum;
+      console.log(`${student.name} failed to grasp the curriculum, and lost ${randomNum} points on their grade!`);
+    }
+  }
+  graduate(student) {
+    if (student.grade >= 70) {
+      console.log(`Congratulations ${student.name}! You graduated!`);
+    } else {
+      console.log(`I'm sorry ${student.name}, you need to go study some more...`);
+    }
+  }
+}
+
+class Student extends Person {
+  constructor(attr) {
+    super(attr);
+    this.previousBackground = attr.previousBackground;
+    this.className = attr.className;
+    this.favSubjects = attr.favSubjects;
+    this.grade = attr.grade;
+    }
+    listsSubjects() {
+      console.log(this.favSubjects);
```

Comment:
The instructions may not be the most clear for this method. You'll want to __iterate__ over the student's favoriteSubjects __one by one__.

Reply:
Alright, changed it :)